app/models/build.rb

```diff
@@ -106,19 +106,25 @@ def validate_git_reference
     return if errors.include?(:git_ref) || errors.include?(:git_sha)
 
     if git_ref.present?
-      commit = project.repository.commit_from_ref(git_ref)
-      if commit
-        self.git_sha = commit unless git_sha.present?
+      if commit = commit_from_ref(git_ref)
+        self.git_sha = commit if git_sha.blank?
```

Comment:
this is actually wrong, we need to validate that the current commit exists like we do in the elsif